crates/client/db/src/mapping_db.rs

```diff
@@ -112,4 +123,25 @@ impl<B: BlockT> MappingDb<B> {
             None => Ok(None),
         }
     }
+
+    /// Returns the list of transaction hashes for the given block hash.
+    ///
+    /// # Arguments
+    ///
+    /// * `starknet_hash` - the hash of the starknet block to search for.
+    ///
+    /// # Returns
+    ///
+    /// The list of transaction hashes.
+    ///
+    /// This function may return `None` for two separate reasons:
+    ///
+    /// - The cache is disabled.
```

Comment:
By adding the `cache` bool to the mappingdb struct directly, you can now perform a check here, and return an error with message if the cache is disable, rather than `None`

Reply:
I don't think that's a good idea to return an error if the cache is disabled. The idea was to abstract the cache as much as possible.

If for some reason the value is not in the cache, we probably want to recompute it rather than returning an error an refusing to respond. And if the cache is disabled, the value will
simply not be in the cache at all. I like those two cases having the same apparent behavior (`None` being returned)

Reply:
Okay to return `None` in both ways. But you should still check the bool, it will avoid you the whole IO reading work

Reply:
oh right. Didn't think about that.
Good catch.